brainscore_language/plugins/pereira2018/ceiling_packaging.py

```diff
@@ -0,0 +1,305 @@
+                score = score.expand_dims(self.subject_column, _apply_raw=apply_raw)
+                score.__setitem__(self.subject_column, [subject], _apply_raw=apply_raw)
+                scores.append(score)
+            except NoOverlapException as e:
+                self._logger.debug(f"Ignoring no overlap {e}")
+                continue  # ignore
+            except ValueError as e:
+                if "Found array with" in str(e):
+                    self._logger.debug(f"Ignoring empty array {e}")
+                    continue
+                else:
+                    raise e
+
+        scores = Score.merge(*scores)
+        scores = apply_aggregate(lambda scores: scores.mean(self.subject_column), scores)
+        return scores
+
+
+class ExtrapolationCeiling:
+    def __init__(self, subject_column='subject', extrapolation_dimension='neuroid',
+                 num_bootstraps=100):
+        self._logger = logging.getLogger(fullname(self))
+        self.subject_column = subject_column
+        self.extrapolation_dimension = extrapolation_dimension
+        self.num_bootstraps = num_bootstraps
+        self.num_subsamples = 10
+        self.holdout_ceiling = HoldoutSubjectCeiling(subject_column=subject_column)
+        self._rng = RandomState(0)
+
+    def __call__(self, assembly, metric):
+        scores = self.collect(identifier=assembly.identifier, assembly=assembly, metric=metric)
+        return self.extrapolate(identifier=assembly.identifier, ceilings=scores)
+
+    @store(identifier_ignore=['assembly', 'metric'])
+    def collect(self, identifier, assembly, metric):
+        self._logger.debug("Collecting data for extrapolation")
+        subjects = set(assembly[self.subject_column].values)
+        subject_subsamples = tuple(range(2, len(subjects) + 1))
+        scores = []
+        for num_subjects in tqdm(subject_subsamples, desc='num subjects'):
+            for sub_subjects in self._random_combinations(
+                    subjects=set(assembly[self.subject_column].values),
+                    num_subjects=num_subjects, choice=self.num_subsamples, rng=self._rng):
+                sub_assembly = assembly[{'neuroid': [subject in sub_subjects
+                                                     for subject in assembly[self.subject_column].values]}]
+                selections = {self.subject_column: sub_subjects}
+                try:
+                    score = self.holdout_ceiling(assembly=sub_assembly, metric=metric)
+                    score = score.expand_dims('num_subjects')
+                    score['num_subjects'] = [num_subjects]
+                    for key, selection in selections.items():
+                        expand_dim = f'sub_{key}'
+                        score = score.expand_dims(expand_dim)
+                        score[expand_dim] = [str(selection)]
+                    scores.append(score.raw)
+                except KeyError as e:  # nothing to merge
```

Comment:
in the neuralnlp repo, we have had a lot of issues with this error and merge operation (seems like it has been taken care off, but it'd need to actually run it to check -- so flagging this for now)

Reply:
didn't run into it when packaging this time, by default you wouldn't have to rerun this since it's all on S3 now